internal/servers/worker.go

```diff
@@ -4,12 +4,27 @@ import (
 	"context"
 	"strings"
 
+	"github.com/fatih/structs"
 	"github.com/hashicorp/boundary/internal/db/timestamp"
 	"github.com/hashicorp/boundary/internal/errors"
 	"github.com/hashicorp/boundary/internal/servers/store"
 	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/structpb"
 )
 
+type workerAuthState struct {
+	WorkerId string
+}
+
+// AttachWorkerIdToState accepts a workerId and creates a struct
+// This is intended for use in worker authorization; AuthorizeNode accepts the option WithState
+// so that the workerId is passed through to storage
+func AttachWorkerIdToState(workerId string) (*structpb.Struct, error) {
```

Comment:
Would you need to possibly attach version as well?

Reply:
Oh, I see that it's in certAuthState. Although maybe it's just a naming thing but I find it a bit confusing -- certs are generally immutable and workers aren't so it seems odd that cert has version and workers don't?